### PR TITLE
Rankeval: Fold template test project into main module

### PR DESCRIPTION
**Pull request by cbuescher**
This change moves tests in `smoke-test-rank-eval-with-mustache` into the main
ranking evaluation module by declaring that the integration testing cluster
requires the `lang-mustache` plugin. This avoids having to maintain the qa
project for only one basic test suite.

### CODE REVIEW

**Comment by cbuescher** [comment]
@Nik I remember you told be about doing a similar thing with reindex recently and I followed what you did in 51fa8739 mostly. Can you take a quick look? Left one question for my own understanding.

File: modules/rank-eval/build.gradle [hunk 1]
```diff
@@ -23,3 +23,12 @@ esplugin {
   hasClientJar = true
 }
 
+integTestCluster {
+  // Modules who's integration is explicitly tested in integration tests
+  module project(':modules:lang-mustache')
+}
+
+run {
```
Comment (cbuescher) [inline modules/rank-eval/build.gradle]: The IT test where running without addind this run section, however I saw in the commit where you moved the reindex tests that you added them. I'm curious if/when this section is used. Is it for the case when I start clusters from the command line from one of the plugin/module directories or is it used by something else? Or can I safely remove it in my case?

**Comment by nik9000** [comment]
> @nik

Wrong Nik. Poor dude gets my pings every once in a while.

File: modules/rank-eval/build.gradle (hunk 1 above)
Comment (nik9000) [reply modules/rank-eval/build.gradle]: It is used for `./gradlew -p modules/rank-eval run`. I try to keep the `integTestCluster` and `run` configurations the same.

**Review by nik9000 — APPROVED** [review]
LGTM.

**Pull request merged**